lib/Doctrine/ORM/PersistentCollection.php

```diff
@@ -514,10 +514,10 @@ public function offsetGet($offset)
     public function offsetSet($offset, $value)
     {
         if ( ! isset($offset)) {
-            return $this->add($value);
+            $this->add($value);
```

Comment:
Are you calling `offsetSet` directly somewhere? Because from `$collection['foo'] = $value`, you cannot get the return value of `offsetSet` method. I find out about this issue because `offsetSet` has `@return void`, so it should not return anything.